node_modules/col-core/lib/api.js

```diff
@@ -34,12 +34,9 @@ var Server = require('./server');
  * @param  {Object}         callback.err        An error object, if any
  */
 var init = module.exports.init = function(callback) {
-  // All unexpected or uncaught errors will be caught and logged here. At this point we cannot
-  // guarantee that the system is functioning properly anymore so we kill the process. When running
-  // in production, the service script will automatically respawn the instance
+  // All unexpected or uncaught errors will be caught and logged here
   process.on('uncaughtException', function(err) {
-    log.error({'err': err}, 'Uncaught exception was raised, restarting the process');
-    process.exit(1);
```

Comment:
If an error bubbles up to this global catch-all handler the application is pretty much in an unknown state. The recommended practice is to restart your application and start from a clean slate: https://nodejs.org/api/process.html#process_event_uncaughtexception

> Note that uncaughtException is a very crude mechanism for exception handling.
> Do not use it as the node.js equivalent of On Error Resume Next. An unhandled exception means your application - and by extension node.js itself - is in an undefined state. Blindly resuming means anything could happen.
> Think of resuming as pulling the power cord when you are upgrading your system. Nine out of ten times nothing happens - but the 10th time, your system is bust.
> 
> You have been warned.

We should strive to never hit this block of logic, when we do, we should probably restart gracefully.